presto-main/src/main/java/com/facebook/presto/operator/SpillableGroupedTopNBuilder.java

```diff
@@ -220,7 +216,7 @@ public void close()
     public ListenableFuture<?> startMemoryRevoke()
     {
         checkState(spillInProgress.isDone());
-        if (inputInMemoryGroupedTopNBuilder.isEmpty()) {
+        if (inputInMemoryGroupedTopNBuilder.isEmpty() || localRevocableMemoryContext.getBytes() == 0) {
```

Comment:
when would the inMemoryGroupedTopNBuilder be not empty but revocable memory be > 0?

Reply:
Checking for `localRevocableMemoryContext` tells us if we are inside the `buildResult` function and inside the `migrateMemoryContext` function, which will mean that the current input is either going to be returned or spilled, so this `startRevokingThread` doesn't need to take any action (specially - do not trigger another spill) of the same input